backend/config/boltdb.go

```diff
@@ -0,0 +1,90 @@
+package config
+
+import (
+	"encoding/json"
+
+	"github.com/boltdb/bolt"
+)
+
+type BoltReaderWriterQuerierImpl struct {
+	bucket  string
+	version string
+	db      *bolt.DB
+}
+
+func NewBoltReader(version string, db *bolt.DB) Reader {
+	return BoltReaderWriterQuerierImpl{
+		bucket:  "ConfigVersion",
+		version: version,
+		db:      db,
+	}
+}
+
+func NewBoltQuerier(db *bolt.DB) Querier {
+	return BoltReaderWriterQuerierImpl{
+		bucket: "ConfigVersion",
+		db:     db,
+	}
+}
+
+func NewBoltWriter(db *bolt.DB) Writer {
+	return BoltReaderWriterQuerierImpl{
+		bucket: "ConfigVersion",
+		db:     db,
+	}
+}
+
+func (b BoltReaderWriterQuerierImpl) Read() (*Config, error) {
+	cfg := &Config{}
+	err := b.db.View(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket([]byte(b.bucket))
+		v := bucket.Get([]byte(b.version))
+		return json.Unmarshal(v, cfg)
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return cfg, nil
+}
+
+func (b BoltReaderWriterQuerierImpl) Write(cfg *Config) error {
+	buff, err := json.Marshal(cfg)
+	if err != nil {
+		return err
+	}
+
+	err = b.db.Update(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket([]byte(b.bucket))
+		if bucket == nil {
+			bucket, err = tx.CreateBucket([]byte(b.bucket))
+			if err != nil {
+				return err
+			}
+		}
+		return bucket.Put([]byte(cfg.Version), buff)
```

Comment:
We should add more meaning for cfg.Version.
example: `version_number:unix_timestamp:checksum`